Add GinkgoTB() function

ebabani opened the pull request:
This function returns a wrapper around GinkgoT() which satisfies the `testing.TB` interface.

- Update fail tests with tests for GinkgoT() and GinkgoTB().
- Fix a bug where GinkgoT().Fail() was not reporting the correct line on error.

onsi commented:
thanks!  this looks great - but I think there’s a bug in the (previous, existing) test that checks that all definitions in the core DSL are exported in the modular DSL package (long story….).  I’ll take a look later today/tomorrow and fix it.

ebabani commented:
> thanks! this looks great - but I think there’s a bug in the (previous, existing) test that checks that all definitions in the core DSL are exported in the modular DSL package (long story….). I’ll take a look later today/tomorrow and fix it.

I pushed a fix for that failure. Forgot to run those tests and was only running integration tests.

I updated the symbol check to ignore functions with a receiver. Those methods are never going to cause an import conflict, and I added the missing package level functions and types to `core_dsl.go`.

onsi commented:
fantastic, thank you! i’ll cut a release soon :)
